### PR TITLE
MULE-17168: Remove softreferences for app classloaders to avoid OOM on redeploy.

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: modules/artifact/src/main/java/org/mule/module/artifact/classloader/ThreadGroupContextClassLoaderSoftReferenceBuster.java [hunk 1]
```diff
@@ -4,28 +4,32 @@
...
+
 /**
  * Reference utilities to bust Thread Group Context soft references.
  * 
  * @since 4.3.0
  */
 public class ThreadGroupContextClassLoaderSoftReferenceBuster {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(ThreadGroupContextClassLoaderSoftReferenceBuster.class);
+  private final transient static Logger logger = LoggerFactory.getLogger(ThreadGroupContextClassLoaderSoftReferenceBuster.class);
```
Comment (elrodro83) [inline modules/artifact/src/main/java/org/mule/module/artifact/classloader/ThreadGroupContextClassLoaderSoftReferenceBuster.java]: why?

File: modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/MuleArtifactClassLoader.java [hunk 2]
```diff
@@ -322,6 +326,8 @@ private void shutdownListeners() {
    */
   protected ResourceReleaser createResourceReleaserInstance() {
     if (resourceReleaserInstance == null) {
+      createClass(threadGroupContextClassLoaderSoftReferenceBusterLocation);
```
Comment (elrodro83) [inline modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/MuleArtifactClassLoader.java]: why is this here? nobody uses the generated class afterwards

Comment (fsgonz) [reply modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/MuleArtifactClassLoader.java]: Because it is used internally by the resourceReleaser.

File: modules/artifact/src/main/java/org/mule/module/artifact/classloader/ThreadGroupContextClassLoaderSoftReferenceBuster.java [hunk 3]
```diff
@@ -4,28 +4,32 @@
...
+
 /**
  * Reference utilities to bust Thread Group Context soft references.
  * 
  * @since 4.3.0
  */
 public class ThreadGroupContextClassLoaderSoftReferenceBuster {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(ThreadGroupContextClassLoaderSoftReferenceBuster.class);
+  private final static Logger logger = LoggerFactory.getLogger(ThreadGroupContextClassLoaderSoftReferenceBuster.class);
```
Comment (elrodro83) [inline modules/artifact/src/main/java/org/mule/module/artifact/classloader/ThreadGroupContextClassLoaderSoftReferenceBuster.java]: uppercase LOGGER
